__tests__/component/form/CreateCustomerForm.test.tsx

```diff
@@ -73,3 +74,14 @@ test('onCancel is invoked', async () => {
 
     await waitFor(() => expect(onCancelSpy.callCount).toEqual(1));
 });
+
+test('a required field cannot be missing', async () => {
+    render(<CreateCustomerForm />);
+    for (const field of Object.keys(testCustomerRequiredFields)) {
+        const customerMissingRequired = _.omit(testCustomerAllFields, field);
+        fillAndSubmitForm(customerMissingRequired as Customer);
+        await waitFor(() => expect(bodySpy.callCount).not.toEqual(1), { timeout: 100 });
```

Comment:
```suggestion
        await waitFor(() => expect(bodySpy.callCount).toEqual(1), { timeout: 100 });
```

Reply:
the point is that this is should always be false. we just wait 100ms for it to be true